### ISSUE TITLE
The resolution.properties not updating chrome version automatically (Win11)

### ISSUE THREAD

**Issue by mNarraMine** [bonigarcia/webdrivermanager#1107]
**Description of the problem**: I updated Chrome to 116. And updated my branch on Win11 machine, launch tests in parallel with Maven (they are on JUnit5 + Cucumber) and got an expection `session not created: This version of ChromeDriver only supports Chrome version 114` for some tests (5 of 20).
I traced where the smoking gun is so the problem was in resolution.properties where parameter chrome116=114.0.5735.90 but should be chrome116=116.0.5845.96.
While driver version is 114.0.5735.90 it downloads Win32 114 driver version and put it into ~\.cache\selenium\chromedriver in parallel with version 116.
If remove clearDriverCache() and edit resolution.properties from chrome116=114.0.5735.90 to chrome116=116.0.5845.96 and delete win32 with 114 driver version then it work untill resolution.properties updates in next launch back to chrome116=114.0.5735.90.

**Browser and version**:  116.0.5845.97 (Official Build) (64-bit)

**Operating system**: Windows 11 (os.version 10.0)

**WebDriverManager version**:  5.4.1

**WebDriverManager call**: WebDriverManager.chromedriver().setup();

**WebDriverManager traces**: Sorry, I have no stacktrace.

**Error log**: `org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114
Current browser version is 116.0.5845.97 with binary path C:\Program Files\Google\Chrome\Application\chrome.exe
Host info: host: 'MS-PC', ip: '192.168.0.247'
Build info: version: '4.11.0', revision: '040bc5406b'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '19.0.2'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [null, newSession {capabilities=[Capabilities {acceptInsecureCerts: true, browserName: chrome, goog:chromeOptions: {args: [--headless=new, --dns-prefetch-disable, --disable-extensions, --allow-insecure-localhost, --no-sandbox, --disable-dev-shm-usage, --disable-notifications, --remote-allow-origins=*, --window-size=1536,864], excludeSwitches: [enable-automation], extensions: []}, pageLoadStrategy: normal}]}]
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:140)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:96)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:68)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:163)
        at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:196)
        at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:171)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:518)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:232)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:159)
        at org.openqa.selenium.chromium.ChromiumDriver.<init>(ChromiumDriver.java:108)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:88)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:83)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:72)
        at ch.hotelplan.base.BaseDriver.createBrowserInstance(BaseDriver.java:92)
        at ch.hotelplan.base.BaseHook.launchBrowser(BaseHook.java:35)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at io.cucumber.java.Invoker.doInvoke(Invoker.java:66)
        at io.cucumber.java.Invoker.invoke(Invoker.java:24)
        at io.cucumber.java.AbstractGlueDefinition.invokeMethod(AbstractGlueDefinition.java:47)
        at io.cucumber.java.JavaHookDefinition.execute(JavaHookDefinition.java:64)
        at io.cucumber.core.runner.CoreHookDefinition.execute(CoreHookDefinition.java:46)
        at io.cucumber.core.runner.HookDefinitionMatch.runStep(HookDefinitionMatch.java:21)
        at io.cucumber.core.runner.ExecutionMode$1.execute(ExecutionMode.java:10)
        at io.cucumber.core.runner.TestStep.executeStep(TestStep.java:84)
        at io.cucumber.core.runner.TestStep.run(TestStep.java:56)
        at io.cucumber.core.runner.TestCase.run(TestCase.java:78)
        at io.cucumber.core.runner.Runner.runPickle(Runner.java:75)
        at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.lambda$runTestCase$4(CucumberEngineExecutionContext.java:112)
        at io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:137)
        at io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)
        at io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:137)
        at io.cucumber.junit.platform.engine.CucumberEngineExecutionContext.runTestCase(CucumberEngineExecutionContext.java:109)
        at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:168)
        at io.cucumber.junit.platform.engine.NodeDescriptor$PickleDescriptor.execute(NodeDescriptor.java:90)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:202)
        at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1841)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)`

**Comment by melloware** [comment]
Seem this stack overflow suggestion the way to download the Chrome Driver and detect its version has changed:  https://stackoverflow.com/a/76925492/502366

**Comment by mNarraMine** [comment]
@melloware  it means that in this way need to code custom driver manager instead of utilising wdm because wdm updateing resolution.properties which return to old version.

**Comment by melloware** [comment]
Yes this library needs to update its code on how it looks up the Chrome Driver version.

**Comment by adrian-herscu** [comment]
Happens on Windows 10 also.

**Comment by decarbonite** [comment]
Even after removing ~/.cache entirely, and rerunning with `WebDriverManager.chromedriver().clearDriverCache().setup()`
the resolution.properties content is still looking for chrome 114.
```#WebDriverManager Resolution Cache
#Mon Aug 21 15:52:20 EDT 2023
chrome=116
chrome-ttl=16\:52\:20 21/08/2023 EDT
chrome116=114.0.5735.90
chrome116-ttl=15\:52\:20 22/08/2023 EDT
```
Is there a resolution for systems on JDK 8 and can't upgrade to 11?

Selenium: 4.11.0
WebDriverManager: 5.3.2

**Comment by dennisFranzCS** [comment]
> Even after removing ~/.cache entirely, and rerunning with `WebDriverManager.chromedriver().clearDriverCache().setup()` the resolution.properties content is still looking for chrome 114.
> 
> ```
> #Mon Aug 21 15:52:20 EDT 2023
> chrome=116
> chrome-ttl=16\:52\:20 21/08/2023 EDT
> chrome116=114.0.5735.90
> chrome116-ttl=15\:52\:20 22/08/2023 EDT
> ```
> 
> Is there a resolution for systems on JDK 8 and can't upgrade to 11?
> 
> Selenium: 4.11.0 WebDriverManager: 5.3.2

For me that didn't work either. My temporary solution is declaring the full version like this:

`
chrome=116
chrome116-ttl=10\:46\:01 23/08/2023 MESZ
chrome-ttl=11\:46\:01 22/08/2023 MESZ
chrome116=116.0.5845.97
`

And then downloading the current webdriver version manually in a folder in ~/.cache/selenium/chromedriver/win32/116.0.5845.97 and deleting all other versions... That is kinda annoying because every miniupdate has to be done manually but temporarely this works for me.

**Comment by bonigarcia** [comment]
I created a new subsection in the [know issues section of the WebDriverManager doc](https://bonigarcia.dev/webdrivermanager/#known-issues) (see _Support for chromedriver 115+_):

In summary, the solution to this problem is to bump WebDriverManager to the **latest version** (5.4.1 currently). Also, to ensure that the wrong version has not been cached in the *resolution cache* (more info on [doc](https://bonigarcia.dev/webdrivermanager/#resolution-algorithm)), you can refresh completely the cache folder (at least once) as follows:

```
WebDriverManager.chromedriver().clearDriverCache().setup();
```

Notice that WebDriverManager 5.4+ requires Java 11 (see [Selenium announcement](https://www.selenium.dev/blog/2023/java-8-support/)).

Also, the [versions.properties](https://github.com/bonigarcia/webdrivermanager/blob/master/src/main/resources/versions.properties) is not used anymore by WebDriverManager to discover chromedriver versions. In fact, it was maintained only for backward compatibility with ancient versions of WebDriverManager. But it is not needed (not updated) anymore.

**Comment by bonigarcia** [comment]
UPDATE: To provide a solution to those that cannot bump to Java 11, I  released WebDriverManager 5.5.0, compiled with Java 8. 

In any case, my personal opinion is that Java 8 is already an ancient version, and I recommend using at least Java 11+ as soon as possible.

**Comment by tyBouch** [comment]
> I created a new subsection in the [know issues section of the WebDriverManager doc](https://bonigarcia.dev/webdrivermanager/#known-issues) (see _Support for chromedriver 115+_).
> 
> In summary, the solution to this problem is to bump WebDriverManager to the **latest version** (5.4.1 currently). Also, to ensure that the wrong version has not been cached in the _resolution cache_ (more info on [doc](https://bonigarcia.dev/webdrivermanager/#resolution-algorithm)), you can refresh completely the cache folder (at least once) as follows:
> 
> ```
> WebDriverManager.chromedriver().clearDriverCache().setup();
> ```
> 
> Notice that WebDriverManager 5.4+ requires Java 11 (see [Selenium announcement](https://www.selenium.dev/blog/2023/java-8-support/)).
> 
> Also, the [versions.properties](https://github.com/bonigarcia/webdrivermanager/blob/master/src/main/resources/versions.properties) is not used anymore by WebDriverManager to discover chromedriver versions. In fact, it was maintained only for backward compatibility with ancient versions of WebDriverManager. But it is not needed (not updated) anymore.

Hi Bonigarcia. I'm using intellij and I'm still running into the issue with 5.4.1

`10:43:42.569 WARN  wdm.WebDriverManager (WebDriverManager.java:815) - Exception deleting driver cache at src\test\resources\drivers
org.apache.commons.io.IOExceptionList: src\test\resources\drivers
	at org.apache.commons.io.FileUtils.cleanDirectory(FileUtils.java:331) ~[commons-io-2.11.0.jar:2.11.0]
	at io.github.bonigarcia.wdm.WebDriverManager.clearDriverCache(WebDriverManager.java:813) [webdrivermanager-5.4.1.jar:?]`
	
`org.openqa.selenium.SessionNotCreatedException: session not created: This version of ChromeDriver only supports Chrome version 114`

`10:43:40.184 WARN  versions.VersionDetector (VersionDetector.java:145) - Exception reading  to get version of chromedriver (no protocol: )
10:43:40.395 ERROR online.HttpClient (HttpClient.java:176) - Error HTTP 404 executing https://chromedriver.storage.googleapis.com/LATEST_RELEASE_116
10:43:40.395 WARN  versions.VersionDetector (VersionDetector.java:162) - Exception reading https://chromedriver.storage.googleapis.com/LATEST_RELEASE_116 to get latest version of chromedriver (Error HTTP 404 executing https://chromedriver.storage.googleapis.com/LATEST_RELEASE_116)
10:43:40.426 WARN  versions.VersionDetector (VersionDetector.java:145) - Exception reading  to get version of chromedriver (no protocol: )
10:43:40.547 INFO  wdm.WebDriverManager (WebDriverManager.java:1564) - Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
10:43:40.831 INFO  online.Downloader (Downloader.java:131) - Downloading https://chromedriver.storage.googleapis.com/114.0.5735.90/chromedriver_win32.zip`
